### PR TITLE
Bump Rubocop to v0.59.0

### LINKED ISSUE #7156: Update dependency constraint to allow for rubocop v0.58.2
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.57.2`, and the latest version available is `0.58.2`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #7235: Update dependency constraint to allow for rubocop v0.59.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.57.2`, and the latest version available is `0.59.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by jekyllbot**
Fix #7235 
Fix #7116  
Fix #7156

### CODE REVIEW

File: lib/jekyll/utils/platforms.rb [hunk 1]
```diff
@@ -70,12 +70,11 @@ def linux?
...
       def proc_version
-        @proc_version ||= begin
-          Pathutil.new(
-            "/proc/version"
-          ).read
-        rescue Errno::ENOENT
-          nil
+        @proc_version ||=
+          begin
+          Pathutil.new("/proc/version").read
```
Comment (ashmaroli) [inline lib/jekyll/utils/platforms.rb]: there should be an indentation here..

Comment (DirtyF) [reply lib/jekyll/utils/platforms.rb]: fixed

**Comment by ashmaroli** [comment]
This should also close #7116 and associated issues when merged..

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev

**Pull request merged**